src/vmcu/services/communications/Socket.py

```diff
@@ -67,3 +74,34 @@ def get_packet(self) -> Optional[bytes]:
     def __del__(self):
         self.stop()
     
+    class Raw_Data_Input(Input):
+        def __init__(self,socket,raw_data):
+            self._socket = socket
+            self._raw_data = raw_data
+        
+        def apply(self):
+            if not self._socket.is_running():
+                raise RuntimeError("Cannot send a tcp message with the socket not running")
```

Comment:
has to extend `InputFailedException` which is in test_lib/input.py, create a custom exception for this error

Reply:
outdated